client/server/middleware/test/logger.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * External dependencies
+ */
+import EventEmitter from 'events';
+
+/**
+ * Internal dependencies
+ */
+import loggerMiddleware from '../logger';
+
+const mockLogger = {
+	info: jest.fn(),
+};
+jest.mock( 'server/lib/logger', () => ( {
+	getLogger: () => mockLogger,
+} ) );
+
+const withRequest = ( { method, url, ip, user, httpVersionMajor, httpVersionMinor, headers } ) => {
+	const req = new EventEmitter();
+	req.method = method;
+	req.url = url;
+	req.ip = ip;
+	req.headers = {
+		...headers,
+	};
+	if ( user ) {
+		req.headers.authorization = 'Basic ' + new Buffer( user + ':pass' ).toString( 'base64' );
+	}
+
+	req.httpVersionMajor = httpVersionMajor;
+	req.httpVersionMinor = httpVersionMinor;
+	return req;
+};
+
+const withResponse = ( { statusCode, headers } ) => {
+	const res = new EventEmitter();
+	res.headersSent = true;
+	res.headers = {
+		...headers,
+	};
+	res.getHeader = ( header ) => res.headers[ header.toLowerCase() ];
+	res.statusCode = statusCode;
+	return res;
+};
+
+const withNodeEnv = ( env ) => {
+	const current = process.env.NODE_ENV;
+	process.env.NODE_ENV = env;
+
+	afterAll( () => {
+		process.env.NODE_ENV = current;
+	} );
+};
+
+beforeAll( () => {
+	jest.useFakeTimers( 'modern' );
```

Comment:
The tests would be better isolated from each other if this was a `beforeEach` method, resetting the fake timers before each test. The [jest docs](https://jestjs.io/docs/en/timer-mocks) recommend that, too.

TIL about the `modern` implementation :+1: We've had some troubles before with Jest fake timers not patching `Date`, and had to use the Sinon implemenation instead.